cashu/lightning/lnd_grpc/lnd_grpc.py

```diff
@@ -0,0 +1,376 @@
+
+    
+    async def create_invoice(
+        self,
+        amount: Amount,
+        memo: Optional[str] = None,
+        description_hash: Optional[bytes] = None,
+        unhashed_description: Optional[bytes] = None,
+        **kwargs,
+    ) -> InvoiceResponse:
+        self.assert_unit_supported(amount.unit)
+        data = lnrpc.Invoice(
+            value=amount.to(Unit.sat).amount,
+            private=True,
+            memo=memo or "",
+        )
+        if kwargs.get("expiry"):
+            data.expiry = kwargs["expiry"]
+        if description_hash:
+            data.description_hash = description_hash
+        elif unhashed_description:
+            data.description_hash = hashlib.sha256(unhashed_description).digest()
+
+        r = None
+        try:
+            async with grpc.aio.secure_channel(self.endpoint, self.combined_creds) as channel:
+                lnstub = lightningstub.LightningStub(channel)
+                r = await lnstub.AddInvoice(data)
+        except AioRpcError as e:
+            logger.error(f"AddInvoice failed: {e}")
+            return InvoiceResponse(
+                ok=False,
+                error_message=f"AddInvoice failed: {e}",
+            )
+
+        payment_request = r.payment_request
+        payment_hash = r.r_hash.hex()
+        checking_id = payment_hash
+
+        return InvoiceResponse(
+            ok=True,
+            checking_id=checking_id,
+            payment_request=payment_request,
+            error_message=None,
+        )
+    
+    async def pay_invoice(
+        self, quote: MeltQuote, fee_limit_msat: int
+    ) -> PaymentResponse:
+        # if the amount of the melt quote is different from the request
+        # call pay_partial_invoice instead
+        invoice = bolt11.decode(quote.request)
+        if invoice.amount_msat:
+            amount_msat = int(invoice.amount_msat)
+            if amount_msat != quote.amount * 1000 and self.supports_mpp:
+                return await self.pay_partial_invoice(
+                    quote, Amount(Unit.sat, quote.amount), fee_limit_msat
+                )
+
+        # set the fee limit for the payment
+        feelimit = lnrpc.FeeLimit(
+            fixed_msat=fee_limit_msat
+        )
+        r = None
+        try:
+            async with grpc.aio.secure_channel(self.endpoint, self.combined_creds) as channel:
+                lnstub = lightningstub.LightningStub(channel)
+                r = await lnstub.SendPaymentSync(
+                    lnrpc.SendRequest(
+                        payment_request=quote.request,
+                        fee_limit=feelimit,
+                    )
+                )
+        except AioRpcError as e:
+            error_message = f"SendPaymentSync failed: {e}"
+            return PaymentResponse(
+                ok=False,
+                error_message=error_message,
+            )
+
+        if r.payment_error and r.payment_error != "":
```

Comment:
Can probably simplified to 
```suggestion
        if r.payment_error:
```